src/scripts/sample/sample.py

```diff
@@ -0,0 +1,156 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""
+Sample benchmark script (lightgbm inferencing)
+"""
+import os
+import sys
+import argparse
+import logging
+import lightgbm
+import numpy
+from distutils.util import strtobool
+
+# Add the right path to PYTHONPATH
+# so that you can import from common.*
+COMMON_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", ".."))
+
+if COMMON_ROOT not in sys.path:
+    print(f"Adding {COMMON_ROOT} to PYTHONPATH")
+    sys.path.append(str(COMMON_ROOT))
+
+# useful imports from common
+from common.metrics import MetricsLogger
+from common.io import input_file_path
+
+
+def get_arg_parser(parser=None):
+    """Adds component/module arguments to a given argument parser.
+
+    Args:
+        parser (argparse.ArgumentParser): an argument parser instance
+
+    Returns:
+        ArgumentParser: the argument parser instance
+
+    Notes:
+        if parser is None, creates a new parser instance
+    """
+    # add arguments that are specific to the script
+    if parser is None:
+        parser = argparse.ArgumentParser(__doc__)
+
+    # recommended: use groups
+    group_i = parser.add_argument_group("I/O Arguments")
+    group_i.add_argument(
+        "--data",
+        required=True,
+        type=input_file_path,  # use this helper type for a directory containing a single file
+        help="Some input location (directory containing a unique file)",
+    )
+    group_i.add_argument(
+        "--model",
+        required=True,
+        type=input_file_path,  # use this helper type for a directory containing a single file
+        help="Some input location (directory containing a unique file)",
+    )
+    group_i.add_argument(
+        "--output",
+        required=True,
+        default=None,
+        type=str,
+        help="Some output location (directory)",
+    )
+
+    group_general = parser.add_argument_group("General parameters")
+    group_general.add_argument(
+        "--verbose",
+        required=False,
+        default=False,
+        type=strtobool,  # use this for bool args, do not use action_store=True
+        help="set True to show DEBUG logs",
+    )
+    group_general.add_argument(
```

Comment:
the goal of this argument is to let the orchestrator pass whatever can be recorded as properties or metrics from the outside of the script (VM type, number of nodes, benchmark name, etc)